src/lib/Scenes/Home/Components/ForYou/index.tsx

```diff
@@ -37,58 +40,96 @@ export class ForYou extends React.Component<Props, State> {
   }
 
   componentDidMount() {
+    const { forYou } = this.props
     const rows: DataSourceRow[] = []
-    const artworkModules = this.props.forYou.artwork_modules || []
-    const artistModules = this.props.forYou.artist_modules && this.props.forYou.artist_modules.concat()
-    const fairsModule = this.props.forYou.fairs_module
-    rows.push({ type: "fairs", data: fairsModule })
-    for (let i = 0; i < artworkModules.length; i++) {
-      const artworkModule = artworkModules[i]
-      rows.push({ type: "artwork", data: artworkModule })
-      if ((i + 1) % 2 === 0) {
+    const artworkModules = forYou.artwork_modules || []
+    const artistModules = forYou.artist_modules && forYou.artist_modules.concat()
+    const fairsModule = forYou.fairs_module
+
+    rows.push({
+      type: "fairs",
+      data: fairsModule,
+    })
+
+    artworkModules.forEach((artworkModule, index) => {
+      rows.push({
+        type: "artwork",
+        data: artworkModule,
+      })
+
+      const alternateRow = (index + 1) % 2 === 0
+
+      if (alternateRow) {
         const artistModule = artistModules.shift()
         if (artistModule) {
-          rows.push({ type: "artist", data: artistModule })
+          rows.push({
+            type: "artist",
+            data: artistModule,
+          })
         }
       }
-    }
+    })
+
     this.setState({
       dataSource: this.state.dataSource.cloneWithRows(rows),
     })
   }
 
-  handleRefresh = () => {
-    this.setState({ isRefreshing: true })
-    const stopRefreshing = () => this.setState({ isRefreshing: false })
-
-    Promise.all(
-      this.state.modules.map(module => {
-        return new Promise((resolve, reject) => {
-          module.forceFetch(null, readyState => {
-            if (readyState.error) {
-              reject(readyState.error)
-            } else if (readyState.aborted) {
-              reject()
-            } else if (readyState.done) {
-              resolve()
-            }
+  handleRefresh = async () => {
+    this.setState({
+      isRefreshing: true,
+    })
+
+    try {
+      await Promise.all(
+        this.state.modules.map(module => {
+          return new Promise((resolve, reject) => {
+            const { props, state: { data, relayProp } } = module
+
+            relayProp.refetch({ ...props.rail, fetchContent: true }, null, error => {
+              if (error) {
+                reject({ error })
+              } else {
+                resolve()
+              }
+            })
           })
         })
+      )
+
+      // Success
+      this.setState({
+        isRefreshing: false,
+      })
+
+      // Fail
+    } catch (error) {
+      this.setState({
+        error, // TODO: Display this somehow
```

Comment:
Not sure if there's a global "toast" error handler / how ya'll are dealing with errors but ^ needs to be wired up. Currently just a stub for what would potentially be an array of errors returned from each rail.

Reply:
Yeah, we're currently not doing anything useful with these errors in terms of displaying them to the user. Wasn't sure what that would look like as we don't have a design for it yet.

Reply:
If this could indicate an unexpected exception then make sure to call `console.error(error)` so it gets reported to Sentry.

Reply:
We do have some informative error UI in Eigen, we should revisit hooking something like that up.